cmake/ElektraUninstall.cmake

```diff
@@ -28,3 +33,78 @@ foreach (file ${files})
 		message (STATUS "File $ENV{DESTDIR}${file} does not exist.")
 	endif (IS_SYMLINK "$ENV{DESTDIR}${file}" OR EXISTS "$ENV{DESTDIR}${file}")
 endforeach (file)
+
+# ===============
+# = Directories =
+# ===============
+
+function(remove_directories directories)
+	foreach (directory ${directories})
+		set (dir "$ENV{DESTDIR}${directory}")
+		if (EXISTS "${dir}")
+			message(STATUS "Uninstalling directory ${dir}")
+			exec_program (
+				"${CMAKE_COMMAND}"
+				ARGS "-E remove_directory \"${dir}\""
+				OUTPUT_VARIABLE rm_out
+				RETURN_VALUE rm_retval)
+
+			if (NOT "${rm_retval}" STREQUAL 0)
+				message (FATAL_ERROR "Problem when removing ${dir}")
+			endif (NOT "${rm_retval}" STREQUAL 0)
+		endif (EXISTS "${dir}")
+	endforeach (directory ${directories})
+endfunction(remove_directories)
+
+set (DIRECTORIES
+	"/usr/local/include/elektra"
+	"/usr/local/lib/cmake/Elektra"
+	"/usr/local/lib/elektra"
+	"/usr/local/share/doc/elektra"
+	"/usr/local/share/doc/elektra-api"
+	"/usr/local/share/elektra"
+	"/usr/local/share/share/elektra"
+	"/usr/local/lib/python2.7/site-packages/support"
+)
+
+remove_directories ("${DIRECTORIES}")
+
+# The following directories might be empty. The order of the directories is important, since we remove them in the given order.
```

Comment:
Seems to ignore CMAKE_INSTALL_PREFIX?

And it would be bad to remove directories we did not create. Isn't easy to fix though (we do not want to reimplement package managers).

Reply:
> And it would be bad to remove directories we did not create.

Removing empty directories should not be a problem in my opinion. Do you know a scenario where this behavior would cause issues?